File: frontend/src/api/history.ts
import { fetchWithAuth } from "./client";

export type PredictionLog = {
  id: number;
  customer_id: string;
  prediction_type: string;
  churn_probability: number;
  features_json: string | null;
  created_at: string;
};

export type PredictionLogListResponse = {
  items: PredictionLog[];
  total: number;
  page: number;
  size: number;
};

export async function getPredictionHistory(skip = 0, limit = 50, customerId = ""): Promise<PredictionLogListResponse> {
  const params = new URLSearchParams();
  params.append("skip", skip.toString());
  params.append("limit", limit.toString());
  if (customerId) params.append("customer_id", customerId);

  const res = await fetchWithAuth(`/history?${params.toString()}`);
  if (!res.ok) {
    const error = await res.json();
    throw new Error(error.detail || "Failed to fetch history");
  }
  return res.json();
}

File: backend/app/schemas/history.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class PredictionLogResponse(BaseModel):
    id: int
    customer_id: str
    prediction_type: str
    churn_probability: float
    features_json: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True

class PredictionLogListResponse(BaseModel):
    items: list[PredictionLogResponse]
    total: int
    page: int
    size: int
